### DISCUSSION TITLE
NDI Ingest

### DISCUSSION THREAD

**Discussion by fonty37** [OvenMediaLabs/OvenMediaEngine#1331] [Ideas]
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/62656003/dd0be71d-2c70-4e4f-9b0e-7153de490ae5)

**Comment by basisbit** [comment]
(just my 2 cents)
[NDI](https://en.wikipedia.org/wiki/Network_Device_Interface) is a proprietary protocol intended for video transmission over local network. There is no AGPL compliant SDK or library that covers the various NDI protocol variations/flavors, afaik.

vMix however supports output through the SRT protocol, which should cover all use cases. See https://www.vmix.com/help23/index.htm?SRT.html And OvenMediaEngine supports SRT ingest quite well. (I have used SRT ingest of OME for approximately 40 events (which had between 5k and 60k attendees), and consider it to be much more reliable and better than RTMP in every way (except lack of documentation).

For how to set it up in vMix: https://www.streamingvalley.nl/step-by-step-guide-how-to-set-up-srt-in-vmix-for-seamless-live-streaming/
How to set up the ingest side of OME: https://airensoft.gitbook.io/ovenmediaengine/live-source/srt-beta

Am I missing a use case where NDI support would be a better solution?

**Comment by fonty37** [comment]
First of all the NDI SDK IS free to use. https://ndi.video/download-ndi-sdk/
Please see  for Grsteamer https://github.com/teltek/gst-plugin-ndi.

yes you can use vMix  But that only has 4 NDI Outputs If I wnat to take 40 feeds of NDI and convert them all to Web-RTC/SRT Vmix is not the best option for that.
 Be great to have a origin I can feed NDI in to directly or if not SDI.
It really depends on what part of the video production workflow your trying to service? 

If you only just want to do SRT to HLS  look restreamer  has a friendly UI 

SRT Is better that RTMP in many ways. Have you looked in to RIST? 

Have a read I talk about the different streaming protocols in this article I wrote 

https://medium.com/@contact_45426/exploring-first-mile-streaming-protocols-rtmp-rtmps-srt-rist-and-whip-7f29ade98b54

Also have a read of my blog feedback welcome :)

https://medium.com/@contact_45426

**Comment by basisbit** [comment]
Whoever might eventually do the decision about integrating NDI support or not, you might want to read this:
https://www.reddit.com/r/VIDEOENGINEERING/comments/wly4r9/time_for_ffmpeg_to_let_ndi_back_in/

Seems like it might be better to not touch NDI / NewTek not even with a 10 foot pole.

**Reply by fonty37** [reply]
Yes it doenst work natavily with FFMPEG but is does with Gstreamer......

**Comment by krisdahl** [comment]
I do think there are use cases for Oven Media Engine on premises with NDI. We would deploy on premises for this workflow on certain scenarios.
